test/net/sourceforge/kolmafia/session/BanishManagerTest.java

```diff
@@ -0,0 +1,365 @@
+package net.sourceforge.kolmafia.session;
+
+import static internal.helpers.Preference.isSetTo;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.arrayContaining;
+import static org.hamcrest.Matchers.arrayWithSize;
+import static org.hamcrest.Matchers.nullValue;
+import static org.junit.jupiter.api.Assertions.*;
+
+import net.sourceforge.kolmafia.KoLCharacter;
+import net.sourceforge.kolmafia.combat.MonsterStatusTracker;
+import net.sourceforge.kolmafia.persistence.MonsterDatabase;
+import net.sourceforge.kolmafia.preferences.Preferences;
+import net.sourceforge.kolmafia.request.StandardRequest;
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.ValueSource;
+
+class BanishManagerTest {
+  @BeforeEach
+  private void beforeEach() {
+    KoLCharacter.reset("BanishManagerTest");
+    Preferences.reset("BanishManagerTest");
+    BanishManager.clearCache();
+  }
+
+  @AfterAll
+  private static void cleanup() {
+    BanishManager.clearCache();
+  }
+
+  @Test
+  void clearCache() {
+    KoLCharacter.setCurrentRun(128);
+    Preferences.setString("banishedMonsters", "fluffy bunny:Be a Mind Master:119");
+    BanishManager.loadBanishedMonsters();
+
+    assertTrue(BanishManager.isBanished("fluffy bunny"));
+    BanishManager.clearCache();
+
+    assertFalse(BanishManager.isBanished("fluffy bunny"));
+  }
+
+  @Test
+  void loadBanishedMonsters() {
+    KoLCharacter.setCurrentRun(128);
+    Preferences.setString(
+        "banishedMonsters",
+        "gingerbread lawyer:snokebomb:118:unhinged survivor:Feel Hatred:119:grizzled survivor:Reflex Hammer:119:cat-alien:mafia middle finger ring:119:alielf:v for vivala mask:119:whiny survivor:stinky cheese eye:119:crate:louder than bomb:119:fluffy bunny:Be a Mind Master:119:paper towelgeist:divine champagne popper:128");
+    BanishManager.loadBanishedMonsters();
+
+    assertTrue(BanishManager.isBanished("gingerbread lawyer"));
+    assertTrue(BanishManager.isBanished("unhinged survivor"));
+    assertTrue(BanishManager.isBanished("grizzled survivor"));
+    assertTrue(BanishManager.isBanished("cat-alien"));
+    assertTrue(BanishManager.isBanished("alielf"));
+    assertTrue(BanishManager.isBanished("whiny survivor"));
+    assertTrue(BanishManager.isBanished("crate"));
+    assertTrue(BanishManager.isBanished("fluffy bunny"));
+    assertTrue(BanishManager.isBanished("paper towelgeist"));
+
+    assertFalse(BanishManager.isBanished("zmobie"));
+  }
+
+  @Test
+  void saveBanishedMonsters() {
+    assertThat("banishedMonsters", isSetTo(""));
+
+    // This will be removed because its run out.
```

Comment:
`it's`, as in `it has`. Or just in long.

Reply:
Oh yeah, I'll fix this next round